drivers/adc/adc_cc23x0.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright (c) 2024 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc23x0_adc
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_cc23x0, CONFIG_ADC_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+
+#include <driverlib/adc.h>
+#include <driverlib/clkctl.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define CPU_FREQ DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
+
+#define ADC_CC23_CH_UNDEF	0xff
```

Comment:
I think I would prefer CC23X0 instead of just CC23. It matches the file name. Same for the other defines below.
```suggestion
#define ADC_CC23X0_CH_UNDEF	0xff
```